plugin.js

```diff
@@ -1,16 +1,16 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const fastUri = require('fast-uri')
 
 function fastifyUrlData (fastify, options, next) {
   fastify.decorateRequest('urlData', function (key) {
     const scheme = this.headers[':scheme'] ? this.headers[':scheme'] : this.protocol
     const host = this.hostname
     const path = this.headers[':path'] || this.raw.url
-    const urlData = fastUri.parse(scheme + '://' + host + path)
-    if (key) return urlData[key]
-    return urlData
+    const url = new URL(scheme + '://' + host + path)
+
+    if (key) return url[key]
+    return url
```

Comment:
can you avoid the breaking change?

Reply:
To avoid it, would mean converting the `URL` to an `URIComponent` object. 
I can map it 1-to-1 and return what we previously had, but there's a s mall caveat where `fast-uri` previously also had a prop on the `URIComponent` called `userinfo` which returned something like this : `"user:pass"`. 
By default the `URL` does not have it, it would require creating ourselves just like as `fast-uri` does it.

Reply:
nevermind